gpt_index/query_engine/types.py

```diff
@@ -0,0 +1,15 @@
+from dataclasses import dataclass
+from typing import Optional
+from gpt_index.indices.query.base import BaseQueryEngine
+
+
+@dataclass
+class Metadata:
```

Comment:
out of curiosity what's your thought about relationship between this + our node objects

Reply:
I think this deserves some deeper thought down the road.

Right now, this metadata is mostly transient (it's only help in the caller layer) and not persisted to storage.

Reply:
ok sounds good